NString.Tests/StringTemplateConverterTests.cs

```diff
@@ -0,0 +1,335 @@
+﻿using System.Globalization;
+using System.Linq;
+using Xunit;
+// ReSharper disable InconsistentNaming
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace NString.Tests
+{
+    public class StringTemplateConverterTests
+    {
+        private static readonly string[] StringCollectionValues = { "1", "2", "3" };
+        private const string ExpectedStringCollectionConcatenation = "1|2|3";
+
+        [Fact]
+        public void StringTemplateValueConverter_Can_Convert_Property()
+        {
+            var values = new PropertyOrFieldValues { ConcatenatedStringCollectionProperty = StringCollectionValues };
+
+            string actual = StringTemplate.Format("{ConcatenatedStringCollectionProperty}", values);
+
+            Assert.Equal(ExpectedStringCollectionConcatenation, actual);
+        }
+
+        [Fact]
+        public void StringTemplateValueConverter_Can_Convert_Field()
+        {
+            var values = new PropertyOrFieldValues { ConcatenatedStringCollectionField = StringCollectionValues };
+
+            string actual = StringTemplate.Format("{ConcatenatedStringCollectionField}", values);
+
+            Assert.Equal(ExpectedStringCollectionConcatenation, actual);
+        }
+
+        [Fact]
+        public void StringTemplateValueConverter_Can_Convert_Reference_Type()
+        {
+            var values = new ReferenceOrValueTypeValues { ConcatenatedStringCollection = StringCollectionValues };
+
+            string actual = StringTemplate.Format("{ConcatenatedStringCollection}", values);
+
+            Assert.Equal(ExpectedStringCollectionConcatenation, actual);
+        }
+
+        [Fact]
+        public void StringTemplateValueConverter_Can_Convert_Value_Type()
+        {
+            var values = new ReferenceOrValueTypeValues { DateTimeToTicks = new DateTime(ticks: 42) };
+
+            string actual = StringTemplate.Format("{DateTimeToTicks}", values);
+
+            Assert.Equal("42", actual);
+        }
+
+        [Fact]
+        public void StringTemplateValueConverter_Works_With_FormatProvider()
+        {
+            var values = new ValuesForFormatProvider { ConvertedWithFortyTwoDaysAdded = DateTime.Today };
+            var expectedConvertedDate = values.ConvertedWithFortyTwoDaysAdded + TimeSpan.FromDays(42);
+
+            var cultures = new[] { CultureInfo.InvariantCulture, null, new CultureInfo("fr-FR") };
+
+            foreach (var culture in cultures)
+            {
+                string expected = expectedConvertedDate.ToString("D", culture);
+                string actual = StringTemplate.Format("{ConvertedWithFortyTwoDaysAdded:D}", values, true, culture);
+                Assert.Equal(expected, actual);
+            }
+        }
+
+        [Fact]
+        public void StringTemplateValueConverter_Works_With_FormatProvider_And_Return_Type_Change()
```

Comment:
This test doesn't actually specify a format provider

Reply:
now it does!